test/c3/appframework/appframework_test.go

```diff
@@ -211,4 +212,102 @@ var _ = Describe("c3appfw test", func() {
 			testenv.VerifyAppsCopied(deployment, testenvInstance, testenvInstance.GetName(), podNames, appListV1, false, true)
 		})
 	})
+
+	Context("Clustered deployment (C3 - clustered indexer, search head cluster)", func() {
+		It("c3, appframework: can deploy a C3 SVA and have ES app installed on SHC", func() {
+
+			// ES is a huge file, we configure it here rather than in BeforeSuite/BeforeEach to save time for other tests
+			// Upload ES app to S3
+			esApp := "SplunkEnterpriseSecuritySuite"
+			appListV1 = append(testenv.BasicApps, esApp)
+			appFileList := testenv.GetAppFileList(appListV1, 1)
+
+			// Download V1 Apps from S3
+			err := testenv.DownloadFilesFromS3(testDataS3Bucket, s3AppDirV1, downloadDirV1, appFileList)
+			Expect(err).To(Succeed(), "Unable to download V1 app files")
+
+			// Upload V1 apps to S3
+			uploadedFiles, err := testenv.UploadFilesToS3(testS3Bucket, s3TestDir, appFileList, downloadDirV1)
```

Comment:
BeforeEach is also downloading and then uploading basic apps. This is repetitive.
```
// Upload V1 apps to S3
s3TestDir = "c3appfw-" + testenv.RandomDNSName(4)
appFileList := testenv.GetAppFileList(appListV1, 1)
uploadedFiles, err := testenv.UploadFilesToS3(testS3Bucket, s3TestDir, appFileList, downloadDirV1)
Expect(err).To(Succeed(), "Unable to upload apps to S3 test directory")
uploadedApps = append(uploadedApps, uploadedFiles...)
```

Reply:
I understand `ES APP` is additional app added to this list. Its better to download just that one app instead of downloading and uploading the whole set.